src/Widgets/NotificationsList.vala

```diff
@@ -129,11 +134,23 @@ public class Notifications.NotificationsList : Gtk.ListBox {
 
     private void on_row_activated (Gtk.ListBoxRow row) {
         if (row is NotificationEntry) {
-            unowned NotificationEntry notification_entry = (NotificationEntry) row;
-            notification_entry.notification.run_default_action ();
-            notification_entry.clear ();
+            var notification_entry = (NotificationEntry) row;
+
+            if (notification_entry.notification.default_action != null) {
+                unowned var action_group = get_action_group (ACTION_GROUP_PREFIX);
+                action_group.activate_action (notification_entry.notification.default_action, null);
+            } else if (notification_entry.notification.actions.length > 0) {
+                return;
+            }
```

Comment:
we actually dropped this check in the daemon bubbles, it makes sense to drop it here too.